src/xrpld/ledger/detail/View.cpp

```diff
@@ -1535,6 +1799,59 @@ requireAuth(ReadView const& view, Issue const& issue, AccountID const& account)
     return tesSUCCESS;
 }
 
+TER
+requireAuth(
+    ReadView const& view,
+    MPTIssue const& mptIssue,
+    AccountID const& account)
+{
+    auto const mptID = keylet::mptIssuance(mptIssue.getMptID());
+    auto const sleIssuance = view.read(mptID);
+
+    if (!sleIssuance)
+        return tecOBJECT_NOT_FOUND;
+
+    auto const mptIssuer = sleIssuance->getAccountID(sfIssuer);
+
+    // issuer is always "authorized"
+    if (mptIssuer == account)
+        return tesSUCCESS;
+
+    auto const mptokenID = keylet::mptoken(mptID.key, account);
+    auto const sleToken = view.read(mptokenID);
+
+    // if account has no MPToken, fail
+    if (!sleToken)
+        return tecNO_AUTH;
```

Comment:
Is this state really a problem if the issuance does not require authorization?

Reply:
The payment would be a problem if the account has neither MPTokenIssuance or MPToken object. Although this check may not seem related to what this function is checking, but preferably we should check for this invalid state since its cheap and simple to do. Just my opinion.

Reply:
If an account submits a Payment transaction, with Amount denominated in units of an MPT, and the MPTokenIssuance exists, but the Destination does not have an MPToken for that issuance, then the transaction should fail? Is that right? That's not how trust lines work. Was this a deliberate choice to make MPT different?